### DISCUSSION TITLE
How to connect ZAP (zap-android) to lnd daemon and get the lndconnect scheme right?

### DISCUSSION THREAD

**Discussion by droid192** [lightningnetwork/lnd#6490] [Troubleshooting] [answered]
remote node runs debian lnd-linux-amd64-v0.14.3-beta.tar.gz

the scheme [zap](https://play.google.com/store/apps/details?id=zapsolutions.zap) android asks for
`lndconnect://<host>:<port>?[cert=<base64url DER certifcate>&]macaroon=<base64url macaroon>` can be [qr generated](https://github.com/LN-Zap/lndconnect/blob/master/lnd_connect_uri.md#create-and-show-a-qr-code-using-command-line), but what is the host:port?  can the node_key.onion be used, how?

the tls.cert and admin.maroon are in the lnd datadir.

**Comment by ziggie1984** [comment]
yes you can easily connect also your Node which is running behind tor, for zap-android it uses port 10009 (grpc) so it would look something like this:

`lndconnect://chjlq27t7igzmv66botvnec7lsic3syg7rkplgxq263wc3vbm4qj5ryd.onion:10009.....`

for more infos check this out:

https://github.com/rootzoll/raspiblitz/blob/v1.7/home.admin/config.scripts/bonus.lndconnect.sh#L76

or

https://github.com/LN-Zap/lndconnect/blob/master/lnd_connect_uri.md

**Comment by ziggie1984** [comment]
not sure which node setup your are running but you also need to create a hidden tor grpc service, so you can connect to it.

Because the above onion address, is the address of my hidden service which is listening on port 10009.

**Comment by droid192** [comment, marked as answer]
lnd.conf has no (known) option to expose the grpc port on the tor hidden service v3. 

thus proceed manually in lnd.conf by  https://github.com/lightningnetwork/lnd/issues/5750 
```
listen=localhost
tor.active=true
tor.streamisolation=true
externalip=lndroutekytme3xds6cmbxaniretdgox2hk4cpu4k27jnub3gkfeuhqd.onion:9735
```
and edit torrc  

```
HiddenServiceDir /var/lib/tor/lnd
HiddenServicePort 10009 127.0.0.1:10009 # interface for wallet rpc
HiddenServicePort 9735 127.0.0.1:9735 # interface for p2p network
```
then `sudo systemctl restart tor`  and restart lnd

if you encounter 'problem: Connection failed TLS ALPN negotiation failed with protocols: [h2]' it is due bad mapping 10009 to the lnd announced grpc 127.0.0.1:8080  (which seems logical but errors) instead of to 10009 (rpc). 

voila zap connects.
